nuget/lib/dependabot/nuget/update_checker.rb

```diff
@@ -59,6 +59,9 @@ def up_to_date?
         # is up-to-date
         return true unless requirements_unlocked_or_can_be?
 
+        # If no preferred version can be found then the dependency is up-to-date
+        return true unless preferred_resolvable_version_details
+
```

Comment:
If a dependency can't be found in any of the registries supplied, then I agree with @deivid-rodriguez it should raise an exception that is a "user error" like maybe `DependencyFileNotResolvable` so we can tell the user why their dependency wasn't updated.

If a registry finds the dependency but all versions are filtered out, like it's the latest version, then saying it's up to date is correct. 

Would this suggestion achieve that?

```suggestion
        raise DependencyFileNotResolvable, "#{dependency.name} not found in any registry" if versions.none?

        # If no preferred version can be found then the dependency is up-to-date
        return true unless preferred_resolvable_version_details

```

Reply:
Yeah, that's what I thought was happening here, that no versions could be found at all for the dependency.

However, I'm on the fence about raising this as a user error vs an unexpected error. I initially suggested unexpected error because this could be caused by a Dependabot bug like here. But in general we don't expect `FileParser` to validate that the Dependencies it parses exist, so it seems fine to raise a user error. If it turns out this is an invalid dependency, then we'll probably get a bug report about it.

I wonder if something like this should be pulled to common code, since it seems it could apply to all ecosystems. Not something for this PR though!

Reply:
I prefer a user error in this case because if this does happen and it's not a bug, there's not anything we can do to fix it and it will pollute our error reporting. 

With a user error, the risk is a bug but we will likely get a report for it, and we can still see elevated levels of specific errors in our metrics which we could track down.

Reply:
Yeah, seems fine.

Reply:
I spoke with @deivid-rodriguez offline and here's what we came up with.

It's not uncommon for a package owner to publish a package then immediately de-list it; this might be used to produce dev or daily builds (I've done this in the past).  In this case the query will return nothing, but when the build requests that specific package it _will_ be returned, so if we start raising `DependencyFileNotResolvable` their entire update scenario will be broken.

The fix I'm going to work on is to update the initial dependency scanning and only report a dependency if one of the package repositories returned _something_ for that dependency.  This will make the dependency scanning pretty network heavy in the beginning, but should ensure that we don't report something that we can't find.